tests/suites/cli/display_clouds.sh

```diff
@@ -111,6 +111,90 @@ EOF
 	fi
 }
 
+run_controller_clouds() {
+	echo
+
+	juju add-cloud my-ec2 -f "./tests/suites/cli/clouds/myclouds.yaml" --force --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME}
+	OUT=$(juju clouds --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME} --format=json 2>/dev/null | jq '.[]')
+
+	EXPECTED=$(
+		cat <<'EOF'
+{
+  "defined": "public",
+  "type": "ec2",
+  "auth-types": [
+    "access-key"
+  ],
+  "regions": {
+    "us-west-1": {
+      "endpoint": "https://ec2.us-west-1.amazonaws.com"
+    },
+    "us-west-2": {
+      "endpoint": "https://ec2.us-west-2.amazonaws.com"
+    }
+  },
+  "users": {
+    "admin": {
+      "display-name": "admin",
+      "access": "admin"
+    }
+  }
+}
+EOF
+	)
+	if [[ ${OUT} != *"${EXPECTED}"* ]]; then
```

Comment:
If I'm reading this right, we're testing that $OUT *contains* $EXPECTED. Shouldn't we be testing for exact equality instead? If not, maybe a comment explaining why this is okay.

Reply:
the controller can have multiple clouds; we can only reasonable check the new one we added